src/parser.ts

```diff
@@ -0,0 +1,261 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+//
+// A generic text parsing class for VexFlow.
+
+import { Vex } from './vex';
+import { Grammar } from './easyscore';
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// TODO: Move to easyscore.ts
+type RuleFunction = () => Rule;
+type TriggerFunction = (state?: Result) => void;
+interface Rule {
+  // Lexer Rules
+  token?: string; // The token property is a string that is compiled into a RegExp.
+  noSpace?: boolean; // TODO: None of the EasyScore rules specify noSpace, so it is not used anywhere.
+
+  // Parser Rules
+  expect?: RuleFunction[];
+  zeroOrMore?: boolean;
+  oneOrMore?: boolean;
+  maybe?: boolean;
+  or?: boolean;
+  run?: TriggerFunction;
+}
+// END: Move to easyscore.ts
+//////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Move these to `types/common`, you will see there other elements that need to be moved. With this we will also probably get git to realise that this is the new version of `parser.js` to facilitate the review.
Did you use `git mv src/parser.js src/parser.ts`? This should also help to realise that this is a new version.

Reply:
Sounds good. The reason I kept it in parser.ts was because I figured I could migrate easyscore.ts next, and it would be a simple edit to bring the types over. Plus, these types would be used only by these two classes. I also tried to avoid editing too many files because I know there are others working on the migration. I don't want to cause too many ugly merges!